oracle/src/services/amqpClient.js

```diff
@@ -24,21 +24,39 @@ function connectWatcherToQueue({ queueName, cb }) {
 }
 
 function connectSenderToQueue({ queueName, cb }) {
+  const deadLetterExchange = `${queueName}-retry`
+
   const channelWrapper = connection.createChannel({
     json: true
   })
 
   channelWrapper.addSetup(channel => {
     return Promise.all([
+      channel.assertExchange(deadLetterExchange, 'fanout', { durable: true }),
       channel.assertQueue(queueName, { durable: true }),
+      channel.bindQueue(queueName, deadLetterExchange),
       channel.prefetch(1),
       channel.consume(queueName, msg =>
         cb({
           msg,
           channel: channelWrapper,
           ackMsg: job => channelWrapper.ack(job),
           nackMsg: job => channelWrapper.nack(job, false, true),
-          sendToQueue: data => channelWrapper.sendToQueue(queueName, data, { persistent: true })
+          scheduleForRetry: async (data, msgRetries = 0) => {
+            const retries = msgRetries + 1
+            const delay = retries ** 2 * 1000
+            const retryQueue = `${queueName}-retry-${delay}`
+            await channel.assertQueue(retryQueue, {
+              durable: true,
+              deadLetterExchange,
+              messageTtl: delay,
+              expires: delay * 10
```

Comment:
what difference between `messageTtl` and `expires` is?

Reply:
`messageTtl` is the time in which the message will be marked as dead and redirected to the dead letter Exchange.
`expires` is the time in which the queue will be deleted after disuse. where use means having consumers, being declared (asserted or checked, in this API).

For example with a `delay` value of 1 second:
- The queue will be created (`channel.assertQueue`)
- The message is pushed to the queue
- After one second message will expire and send to dead letter Exchange
- After 10 seconds of created/asserted the queue will be removed.

Reply:
Ok. thanks!